openshift/backend-cron

```diff
@@ -1,7 +1,38 @@
 #!/usr/bin/env ruby
 
-reschedule_jobs_freq = (ENV['RESCHEDULE_JOBS_FREQ'] || 300).to_i
-loop do
-  system('rake', '--trace=stdout', 'reschedule_failed_jobs') or raise 'task crashed'
-  ENV['ONCE'] ? exit : sleep(reschedule_jobs_freq)
+DEFAULT_RESCHEDULE_JOBS_FREQ = 300
+DEFAULT_DELETE_STATS_FREQ = 60*60*24 # 1 day
+
+# The logger used in the delete stats code defaults to /tmp/backend_logger.log
+# When running this executable STDOUT is more convenient to check that
+# everything is working correctly.
+ENV['CONFIG_LOG_PATH'] ||= 'dev/stdout'
+
+def run_rake_task_in_thread(name, freq, check_once = false)
+  Thread.new do
```

Comment:
There is nothing shared between the threads. Have you considered doing a `fork` + `exec`? It would avoid the pitfalls of GIL, although this seems like not much of an opportunity to block.

Reply:
Yes, there's not much blocking. Delete stats has sleeps between `SCAN` calls and the rescheduling only runs a few redis commands every 5 min by default.